lib/bureaucrat/helpers.ex

```diff
@@ -125,24 +125,58 @@ defmodule Bureaucrat.Helpers do
       opts =
         opts
         |> Keyword.put_new(:operation_id, default_operation_id)
+        |> Keyword.update!(:description, &(&1 || default_description()))
+
+      if Keyword.fetch!(opts, :description) != nil do
+        Bureaucrat.Recorder.doc(conn, opts)
+      else
+        file = Keyword.fetch!(opts, :file)
+        line = Keyword.fetch!(opts, :line)
+
+        Mix.shell().info("""
+        The request at #{file}:#{line} won't be recorded by bureaucrat because
+        the description can't be determined and none is explicitly provided.
+        To address this, you can pass the :description option to this macro.
+
+        If this macro is invoked indirectly, via the request macros, such as
+        get and post, you can switch to the _undocumented version to
+        explicitly avoid generating the documentation.
+
+        Alternatively, you can provide the description manually with something
+        like doc(conn, description: "some description"), where conn is the result
+        of the _undocumented macro.
+        """)
+      end
 
-      Bureaucrat.Recorder.doc(conn, opts)
       conn
     end
   end
 
-  def format_test_name("test " <> name), do: name
+  def default_description do
+    # We look for the first test process in the stack traces of this process as well as other callers.
+    # The latter allows us to find the test description even if this function is running in a task started by a test process.
```

Comment:
```suggestion
    # The latter allows us to find the test description even if this function is running 
    # in a private function call or task started by a test process.
```

Would this be a useful note?

Reply:
(the diff isn't showing but i added "in a private function call or"

Reply:
I guess I should clarify this comment. By "callers" i specifically meant the so called caller processes, not functions. The "latter" in the comment refers to these callers, and explains that we need this so we can find the parent test if the request is started from a task.